### PR TITLE
[omnibus] Use the new RPM GPG key to sign RPM packages

### PR DESCRIPTION
**Pull request by KSerrania**
### What does this PR do?

Changes the signing key we use to a new one, and updates the kitchen step-by-step instructions to match the change.

### Motivation

The time has come.

### Additional Notes

In case we're not ready to completely support the new key when we release 6.14, just revert this PR and we'll be fine.

### CODE REVIEW

File: test/kitchen/site-cookbooks/dd-agent-step-by-step/recipes/default.rb [hunk 1]
```diff
@@ -40,6 +40,7 @@
       gpgcheck=1
       repo_gpgcheck=0
       gpgkey=#{protocol}://yum.datadoghq.com/DATADOG_RPM_KEY.public
+             #{protocol}://yum.datadoghq.com/DATADOG_RPM_KEY_E09422B3.public
```
Comment (truthbk) [inline test/kitchen/site-cookbooks/dd-agent-step-by-step/recipes/default.rb]: Just a note to reviewers, this is indeed the correct way to specify multiple GPG keys for a yum repository.

File: omnibus/config/projects/agent.rb [hunk 2]
```diff
@@ -18,8 +18,12 @@
   python_3_embedded "#{install_dir}/embedded3"
   maintainer 'Datadog Inc.' # Windows doesn't want our e-mail address :(
 else
+  if redhat? || suse?
+    maintainer 'Datadog, Inc <package@datadoghq.com>'
```
Comment (truthbk) [inline omnibus/config/projects/agent.rb]: Why this change (just curious)?

**Review by truthbk — APPROVED** [review]
Don't see anything wrong with this, just a small question but shouldn't be a blocker.

File: omnibus/config/projects/agent.rb (hunk 2 above)
Comment (KSerrania) [reply omnibus/config/projects/agent.rb]: When we created our new key, we changed the associated maintainer's name to this. And the signing fails if the `maintainer` declared here is not the same as the one in the key.

**Comment by KSerrania** [comment]
Will add an upgrade note on another PR (forgot to add it here)
